posts/2023-02-20-aws-instant-on.adoc

```diff
@@ -0,0 +1,222 @@
+
+To use the private Amazon ECR registry, first you will need create a repository for the InstantOn image:
+```
+aws ecr create-repository --repository-name getting-started-instanton
+```
+
+In the output of this command, note the `repositoryUri` value. It should match the `<aws_account_id>.dkr.ecr.<region>.amazonaws.com/getting-started-instanton` pattern. For example: `1234567890.dkr.ecr.us-east-1.amazonaws.com/getting-started-instanton`. 
+
+Next, you will need to authenticate to the registry in order to push or pull images using Podman. To do that, use the following command:
+```
+aws ecr get-login-password | sudo podman login --username AWS --password-stdin <aws_account_id>.dkr.ecr.<region>.amazonaws.com
+```
+
+Once authenticated, you can tag and push the InstantOn image to the private ECR registry:
+```
+sudo podman tag dev.local/getting-started-instanton <aws_account_id>.dkr.ecr.<region>.amazonaws.com/getting-started-instanton
+sudo podman push <aws_account_id>.dkr.ecr.<region>.amazonaws.com/getting-started-instanton
+```
+
+=== Provision EKS cluster
+
+We are ready to provision an EKS cluster now. The first step is to create a `cluster.yaml` file with the following contents. 
+
+.cluster.yaml
+[source,yaml]
+----
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: instanton
+  region: us-east-1
+  version: "1.24"
+
+managedNodeGroups:
+  - name: ng-instanton
+    instanceType: t3.micro
+    desiredCapacity: 3
+    amiFamily: Bottlerocket
+    iam:
+       attachPolicyARNs:
+          - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
+          - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
+          - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+          - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+----
+
+The initial `cluster.yaml` file requests a Kuberentes v1.24 cluster to be provisioned with three link:https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html[managed EC2 nodes] that are running the link:https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami-bottlerocket.html[Bottlerocket operating system]. The cluster is just big enough to handle the sample application. If you are planning to deploy other applications to the cluster, adjust the `instanceType` and/or the `desiredCapacity` parameters. Also, make sure that the `region` parameter matches your region. See the link:https://eksctl.io/usage/schema/[config file schema] for additional settings that can be configured when provisioning a cluster.
+
+NOTE: The cluster nodes are created with the EKS optimized Bottlerocket Amazon Machine Image (AMI) instead of the default Amazon Linux AMI. The Bottlerocket AMI as well as the Ubuntu 20 AMI are using a Linux kernel version that supports InstantOn. As of this writing, the Amazon Linux AMI is running a slightly older kernel version that does not fully support InstantOn.
```

Comment:
```suggestion
NOTE: The cluster nodes are created with the EKS optimized Bottlerocket Amazon Machine Image (AMI) instead of the default Amazon Linux AMI. The Bottlerocket AMI and the Ubuntu 20 AMI use a Linux kernel version that supports InstantOn. As of this writing, the Amazon Linux AMI is running a slightly older kernel version that does not fully support InstantOn.
```